src/core/parser_internal/docgen_worker.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "../symbol.hpp"
+#include "../token.hpp"
+#include "../json_writer.hpp"
+#include "../parse_worker.hpp"
+#include "../worker_routine.hpp"
+
+namespace docgen {
+namespace core {
+namespace parser_internal {
+
+using token_class_t = Token<Symbol>;
+
+using writer_class_t = JSONWriter<token_class_t>;
```

Comment:
`writer_t`

Reply:
Will be called `writer_internal_t` for consistency.